flink-table/flink-table-common/src/test/java/org/apache/flink/table/factories/TestCatalogStoreFactory.java

```diff
@@ -19,27 +19,37 @@
 package org.apache.flink.table.factories;
 
 import org.apache.flink.configuration.ConfigOption;
+import org.apache.flink.table.catalog.AbstractCatalogStore;
 import org.apache.flink.table.catalog.CatalogDescriptor;
 import org.apache.flink.table.catalog.CatalogStore;
 import org.apache.flink.table.catalog.exceptions.CatalogException;
 
 import java.util.Collections;
+import java.util.HashMap;
 import java.util.Map;
 import java.util.Optional;
 import java.util.Set;
 
-/** Test catalog store factory for catalog store discovery tests. */
+/**
+ * This class is an implementation of {@link CatalogStoreFactory} for testing purposes. All
+ * instances of {@link CatalogStore} created by the same {@link CatalogStoreFactory} will share the
+ * saved {@link CatalogDescriptor}s.
+ */
 public class TestCatalogStoreFactory implements CatalogStoreFactory {
 
     public static final String IDENTIFIER = "test-catalog-store";
 
+    private Map<String, CatalogDescriptor> sharedDescriptors;
```

Comment:
Looks like a static field is enough if we want to share something crossing multiple instances?